### DISCUSSION TITLE
Snoopy-HSS: changing partition layout

### DISCUSSION THREAD

**Discussion by schlimmchen** [hoylabs/OpenDTU-OnBattery#1068] [Q&A]
Moved from announcement (issue 1025):

# Snoopy-HSS

It's weird.
When I compile for the S3 size is smaller !?
why is there a 2MB limit when i use the generic_S3 ???

Checking size .pio\build\generic_esp32s3\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM: [== ] 24.0% (used 78680 bytes from 327680 bytes)
Flash: [==========] 95.9% (used 1886133 bytes from 1966080 bytes)
Building .pio\build\generic_esp32s3\firmware.bin

In Platform.ini default ENV is still generic_ESP32

Changing partition worked, can also be uploaded Serial via, visual Studio code

# schlimmchen

> It's weird. When I compile for the S3 size is smaller !?

Yes, as the S3 is a slightly different processor with a different instruction set, so the code is can be smaller (more efficient instructions, etc.).

> why is there a 2MB limit when i use the generic_S3 ???

I don't know what you mean. The partition table is the same for ESP32-S3 and non-S3.

# Snoopy-HSS

Flash: [==========] 95.9% (used 1886133 bytes from 1966080 bytes)

Only a little under 2MB availabe.
How can i change, that it uses 16MB filesystem when compiling generic_esp32s3

is this part for all boards?
board_build.partitions = partitions_custom_4mb.csv

it's set in platform.ini to 4MB and ESP32 per default...

**Comment by schlimmchen** [comment]
@Snoopy-HSS Please avoid going off-topic in other issues.

> is this part for all boards?
board_build.partitions = partitions_custom_4mb.csv

Yes.

> How can i change, that it uses 16MB filesystem when compiling generic_esp32s3

You don't. Why do you need that? The new partition layout will use 8MB of flash. It will preserve the littlefs partition (position and size). the app1 partition will go just behind the littlefs partition and will have the same size as app0. You should now be able to edit the partition table CSV yourself.

**Reply by Snoopy-HSS** [reply]
so, i need to changed platform.ini to board_build.partitions = partitions_custom_**16**mb.csv ?

**Reply by schlimmchen** [reply]
Have a look at 2cc086335fe8811e912f47e97925898616b2d8f2, which should answer your questions, and go wild :wink: Remember not to slip unintentional changes to `platformio.ini` in a PR.
